### DISCUSSION TITLE
Theme specific image serving not working.

### DISCUSSION THREAD

**Discussion by sortedcord** [community/community#11602] [Other Feature Feedback, Questions, & Ideas]
Hello, I stumbled accross a blog article where I learnt that we can append `#gh-dark-mode-only` or `#gh-light-mode-only` to make the images visible according the viewer's set theme. I tried doing the same, but it doesn't work for [my profile readme](https://github.com/sortedcord) ?

*Profile May look different now, please take a look at [this commit](https://github.com/sortedcord/sortedcord/tree/c299c46831ea78a984b069c7e94d4a0ccc7eca61)*

```md
<!-- Dark Mode -->
[![Visit my website!](Assets/glitch_header.gif#gh-dark-mode-only)](https://sortedcord.tech)
<!-- Light Mode -->
[![Visit my website!](Assets/glitch_header_light.gif#gh-light-mode-only)](https://sortedcord.tech)
```
This seems to show both the images regardless of the theme I have set.
![image](https://user-images.githubusercontent.com/37407370/154253124-18e3ca3b-c356-46e0-9bbb-9949790b47bd.png)

Is this not supported with other image types as of now? Such as `.gif` or `.svg`? Viewing the readme in Visual Code's markdown viewer does show them properly, which left me quite confused...

**Comment by D1360-64RC14** [comment]
*Eureka!*

```md
<!-- Dark Mode -->
[![Visit my website!](Assets/glitch_header.gif#gh-dark-mode-only)](https://sortedcord.tech#gh-dark-mode-only)
<!-- Light Mode -->
[![Visit my website!](Assets/glitch_header_light.gif#gh-light-mode-only)](https://sortedcord.tech#gh-light-mode-only)
```

[![Visit my website!](https://raw.githubusercontent.com/sortedcord/sortedcord/c299c46831ea78a984b069c7e94d4a0ccc7eca61/Assets/glitch_header.gif#gh-dark-mode-only)](https://sortedcord.tech#gh-dark-mode-only)
[![Visit my website!](https://raw.githubusercontent.com/sortedcord/sortedcord/c299c46831ea78a984b069c7e94d4a0ccc7eca61/Assets/glitch_header_light.gif#gh-light-mode-only)](https://sortedcord.tech#gh-light-mode-only)
For some reason, you need to put the `#gh-light-mode-only` in the image URL **and** in the anchor URL.

I tried putting it only in the link (`https://sortedcord.tech#gh-light-mode-only`) and it worked, but I think it's safer in both. Very strange.

I got it because in the [github-readme-stats was an example](https://github.com/anuraghazra/github-readme-stats#use-githubs-theme-context-tag) with image inside an anchor. I was having trouble trying to make my stats match the color theme.

(To show the banner here, I changed the `Assets/...` with the hard link)
